releases.py

```diff
@@ -26,36 +63,63 @@ def file_as_blockiter(afile, blocksize=65536):
     @staticmethod
     def calculate_sha256(fname):
         try:
-            return ChunkedHash.hash_bytestr_iter(ChunkedHash.file_as_blockiter(open(fname, "rb")), hashlib.sha256())
+            return ChunkedHash.hash_bytestr_iter(ChunkedHash.file_as_blockiter(open(fname, 'rb')), hashlib.sha256())
         except:
             raise
-            return ""
+            return ''
 
 class ReleaseFile():
-    def __init__(self, path, url):
-        self._path = path
-        self._url = url
-        self._output_file = 'releases.json'
+    def __init__(self, args):
+        self._json_file = JSON_FILE
+
+        self._indir = args.input.rstrip(os.path.sep)
+        self._url = args.url.rstrip('/')
+
+        if args.output:
+            self._outdir = args.output.rstrip(os.path.sep)
+        else:
+            self._outdir = self._indir
+
+        self._infile  = os.path.join(self._indir, self._json_file)
+        self._outfile = os.path.join(self._outdir, self._json_file)
+
+        if not os.path.exists(self._indir):
+            raise Exception('ERROR: %s is not a valid path' % self._indir)
 
-        self.display_name = {'Generic.x86_64': 'Generic AMD/Intel/NVIDIA (x86_64)',
+        if not os.path.exists(self._outdir):
+            raise Exception('ERROR: %s is not a valid path' % self._outdir)
+
+        self._regex_custom_sort = re.compile(r'([0-9]+)\.([0-9]+)\.([0-9]+)')
+        self._regex_train = re.compile(r'([0-9]+.[0-9]+)')
+        self._regex_builds = re.compile(r'LibreELEC-([^-]*)-.*')
+
+        self.display_name = {'A64.arm': 'Allwinner A64',
+                             'AMLG12.arm': 'Amlogic G12A/G12B/SM1',
+                             'AMLGX.arm': 'Amlogic GXBB/GXL/GXM',
+                             'Dragonboard.arm': 'Qualcomm Dragonboard',
+                             'Generic.x86_64': 'Generic AMD/Intel/NVIDIA (x86_64)',
+                             'H3.arm': 'Allwinner H3',
+                             'H6.arm': 'Allwinner H6',
                              'imx6.arm': 'Cubox-i2/i4 and Hummingboard (iMX6)',
-                             'KVIM.arm': 'Khadas Vim',
-                             'KVIM2.arm': 'Khadas Vim 2',
-                             'LePotato.arm': 'LePotato (AML-S905X-CC)',
-                             'MiQi.arm': 'Rockchip MiQi RK3328',
+                             'iMX6.arm': 'Cubox-i2/i4 and Hummingboard (iMX6)',
```

Comment:
two iMX6 in different cases intentional?

Reply:
Yes, old imx6 and new iMX6... :man_shrugging: